src/inmanta/deploy/timers.py

```diff
@@ -0,0 +1,255 @@
+"""
+    Copyright 2024 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+import asyncio
+import logging
+import sys
+from collections.abc import Coroutine
+from typing import Any, Callable
+
+import inmanta.deploy.scheduler
```

Comment:
> I haven't looked a the rest of the file yet, so I may still be proven wrong.

I have by now, and as discussed in our meeting, I'm OK with it being passed as an argument, considering that the alternatives are of the same degree of ugliness. I do stand by the second comment though.

Reply:
Oops, didn't mean to close this.

> typing-only import

Do you mean like so ?:
```
if TYPE_CHECKING:
    from inmanta.deploy.scheduler import ResourceScheduler
```

> use inmanta.util as util instead of importing its objects?

:+1:

Reply:
> Do you mean like so ?:

The `TYPE_CHECKING` part for sure. The unqualified import I don't care about too much, one way or the other.